pyproject.toml

```diff
@@ -34,18 +34,26 @@ fawltydeps = "fawltydeps.main:main"
 [tool.poetry.dependencies]
 # These are the main dependencies for fawltydeps at runtime.
 # Do not add anything here that is only needed by CI/tests/linters/developers
-python = "^3.7.2"
-importlib_metadata = "=6.6.0"
+python = ">=3.7.2"
+importlib_metadata = [
+    # v6.8.0 drops support for Python v3.7:
+    {version = ">=6.6.0", python = ">=3.8"},
+    {version = ">=6.6.0,<6.8.0", python = "<3.8"},
+]
 isort = [
-    # isort 5.12.0 drops support for Python v3.7:
+    # v5.12.0 drops support for Python v3.7:
     {version = "^5.10", python = ">=3.8"},
     {version = ">=5.10,<5.12.0", python = "<3.8"},
 ]
-pip-requirements-parser = "^32.0.1"
+pip-requirements-parser = ">=32.0.1"
```

Comment:
Comment: It looks like the authors of this library do not follow semantic versioning pattern `MAJOR.MINOR.PATCH` unless they started with `MAJOR=30`, see https://pypi.org/project/pip-requirements-parser/#history.

Reply:
Yeah, and this is a bigger problem (IMHO) with Poetry's affinity for using `^` by default: It assumes that packages are (a) using SemVer, and (b) using it correctly. My impression so far is that a lot of packages fail either (a) or (b), and even then, what constitutes a compatibility-breaking change will differ dependending on _how_ you are using the package, and a breaking change might not even be discovered until it's too late (i.e. after release).

As we discussed previously today, I think we're better off being more liberal/flexible in our dependency version specifications, but rather have automatic tests that will quickly flag problems when a new version of a dependency is released.